### DISCUSSION TITLE
RDFLib 8 Scope

### DISCUSSION THREAD

**Discussion by nicholascar** [RDFLib/rdflib#2995] [Ideas]
This Discussion is about the high-level, major, items of functionality in RDFLib that are new, or to be altered, in RDFLib 8. Non-breaking changes could be folded into a 7.x release.

Here are a list of the potential items, as discussed on the RDFLib mailing list in Oct/Nov 2024, with expected work times and people against them:

Item | Breaking | Dec | Jan | Feb | Mar | Who
--- | --- | --- | --- | --- | --- | ---
1 Python 3.11+ only | yes | x | &nbsp; | &nbsp; | &nbsp; | David
2 Build system improvements | no | &nbsp; | x | &nbsp; | &nbsp; | Ashley, Edmond
3 Updates I/O functions | yes | x | x | &nbsp; | &nbsp; | Nick, Ashley, Edmond
4 Updates Store interface | yes | x | x | &nbsp; | &nbsp; | Ashley, David
5 Improved listing of known stores | no | x | &nbsp; | &nbsp; | &nbsp; | Nick, Ashley, Alex
6 SQLite (SQLAlchemy) Store | no | &nbsp; | &nbsp; |  x | &nbsp; | Alex
7 Modern K/V Store | no | &nbsp; | &nbsp; |  x | &nbsp; | Nick
8 Updated Dataset & Graph API | yes | &nbsp; | x | &nbsp; | &nbsp; | Ashley
9 Updated Turtle Parser/Serliazer | yes/no | &nbsp; | x | &nbsp; | &nbsp; | Edmond
10 Updates to Namespace Manager | yes | &nbsp; | &nbsp; | x | &nbsp; | Edmond
11 Updated DefinedNamespace instnaces | no | x | &nbsp; | &nbsp; | &nbsp; | Nick
11 Improved documentation | no | &nbsp; | &nbsp; | x | &nbsp; | Nick

Other proposed changes which are not critical for an initial 8 release are:

Item | Breaking 
--- | ---
x RDF-star handling | no 
y improved SPARQL handing | no

**Comment by jnicholls82** [comment]
Hello,

RDF-star/SPARQL-star support would be really, really appreciated!

Thanks,
James

**Reply by namedgraph** [reply]
I think that's a stretch. Even the current SPARQL execution is super slow.

**Reply by jnicholls82** [reply]
> I think that's a stretch. Even the current SPARQL execution is super slow.

Is there any realistic timeframe for RDF*?

**Reply by nicholascar** [reply]
I think we have to prioritise an 8 release in September without RDF 1.2, because we have a long list of small but getting critical fixes. We'll then have to tackle 1.2 as a whole major task. SPARQL performance might also be another major thing, but clearly 1.2 handing should come first!
